app/sentinel/internal/safe/get.go

```diff
@@ -26,33 +26,52 @@ import (
 )
 
 func acquireSource(ctx context.Context) (*workloadapi.X509Source, bool) {
-	source, err := workloadapi.NewX509Source(
-		ctx, workloadapi.WithClientOptions(workloadapi.WithAddr(env.SpiffeSocketUrl())),
-	)
+	resultChan := make(chan *workloadapi.X509Source)
+	errorChan := make(chan error)
 
-	if err != nil {
-		fmt.Println("Post: I cannot execute command because I cannot talk to SPIRE.")
-		fmt.Println("")
-		return nil, false
-	}
+	go func() {
```

Comment:
The `workloadapi.NewX509Source()` was blocking if there was a network partition.

Handling it in its own goroutine, so that if we get a timeout or `SIGINT` we can honor that.

Reply:
Might make sense to extract this into its own function; but, I’m leaving that to a follow-up commit.